aopy/analysis/connectivity.py

```diff
@@ -0,0 +1,238 @@
+import numpy as np
+from tqdm.auto import tqdm
+import multiprocessing as mp
+
+from . import base
+from .. import data as aodata
+from .. import precondition
+from . import accllr
+
+def get_acq_ch_near_stimulation_site(stim_site, stim_layout='Opto32', electrode_layout='ECoG244', 
+                                 dist_thr=1, return_idx=False):
+    '''
+    Get acquisition channels near a stimulation site. Warning: this function returns channel
+    numbers, which sometimes are 1-indexed. Set the return_idx flag to True to get the channel
+    indices as well.
+
+    Args:
+        stim_site: int, stimulation site
+        stim_layout: str, layout of stimulation sites
+        electrode_layout: str, layout of electrodes
+        dist_thr: float or tuple, threshold for distance from stimulation site
+
+    Returns:
+        acq_ch: np.ndarray, acquisition channels near stimulation site
+        | or **(acq_ch, idx) tuple**, if return_idx is True
+    '''
+    elec_pos, acq_ch, _ = aodata.load_chmap(electrode_layout)
+    stim_pos, stim_ch, _ = aodata.load_chmap(stim_layout)
+    stim_site_pos = stim_pos[stim_ch == stim_site]
+    if stim_site_pos.size == 0:
+        raise ValueError(f"Stimulation site {stim_site} not found in layout {stim_layout}")
+    
+    dist = np.linalg.norm(elec_pos - stim_site_pos, axis=1)
+    try:
+        idx = (dist < dist_thr[1]) & (dist >= dist_thr[0])
+    except:
+        idx = dist < dist_thr
```

Comment:
I think these should be <= also. Also, I think using a try/except statement is a little too general. I think it should either:
1. Use isinstance(dist_thr, tuple)/isinstance(dist_thr, float)
2. Have a way to tell the user which approach is used.

Reply:
i want to make it so that if you ask for dist_thr=[0,1] and dist_thr=[1,2] you get exclusively different results

Reply:
i made it try / except TypeError, is that better? what specifically are you worried about when you say it's too general?

Reply:
In general I just don't like try/excepts because they can mask errors/bugs in the code. The user may expect that the initial approach is working, but if there is an unrelated error the function will just return the wrong answer without the user knowing. 

I prefer to condition on what is actually important. Its hard to think of an example because the error would be an unknown unknown.

Reply:
> i want to make it so that if you ask for dist_thr=[0,1] and dist_thr=[1,2] you get exclusively different results

Sounds good!